K2Field.K2NE.ServiceBroker/Helpers/PowerShell/PowerShellHelper.cs

```diff
@@ -0,0 +1,133 @@
+                    {
+                        powerShellInstance.Runspace.SessionStateProxy.SetVariable(variable.Name, variable.Value);
+                    }                    
+
+                    powerShellInstance.AddScript(powerShellScript);
+                    
+                    // begin invoke execution on the pipeline
+                    Collection<System.Management.Automation.PSObject> returnValue = powerShellInstance.Invoke();
+
+                    //get input variables
+                    foreach (PowerShellVariablesDC variable in variablesList)
+                    {
+                        variable.Value = powerShellInstance.Runspace.SessionStateProxy.GetVariable(variable.Name);
+                    }
+
+                    return GetScriptOutput(returnValue);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        private static string GetScriptOutput(Collection<System.Management.Automation.PSObject> psObjectValues)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            foreach (System.Management.Automation.PSObject obj in psObjectValues)
+            {
+                stringBuilder.AppendLine(obj.ToString());
+            }
+            return stringBuilder.ToString();
+        }
+
+        #region Directory interaction
+        
+        public static string LoadScriptByPath(string filePath)
+        {
+            try
+            {
+                // Create an instance of StreamReader to read from our file. 
+                // The using statement also closes the StreamReader. 
+                using (StreamReader sr = new StreamReader(filePath))
+                {
+
+                    // use a string builder to get all our lines from the file 
+                    StringBuilder fileContents = new StringBuilder();
+
+                    // string to hold the current line 
+                    string curLine;
+
+                    // loop through our file and read each line into our 
+                    // stringbuilder as we go along 
+                    while ((curLine = sr.ReadLine()) != null)
+                    {
+                        // read each line and MAKE SURE YOU ADD BACK THE 
+                        // LINEFEED THAT IT THE ReadLine() METHOD STRIPS OFF 
+                        fileContents.Append(curLine + "\n");
+                    }
+
+                    // call RunScript and pass in our file contents 
+                    // converted to a string 
+                    return fileContents.ToString();
+                }
+            }
+            catch (Exception e)
+            {
+                // Let the user know what went wrong. 
+                string errorText = "The file could not be read:";
+                errorText += e.Message + "\n";
+                throw new Exception(errorText);
```

Comment:
*always* add the original exception to this, so that up the chain we can look for the innerexception. This helps a lot to find stacktraces that are meaningful.